.github/workflows/publish.yml

```diff
@@ -0,0 +1,76 @@
+name: Build & publish module to registry
+
+on:
+  release:
+    types: [published]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        include:
+          - platform: linux/amd64
+            os: ubuntu-latest
+            container:
+              image: ghcr.io/viamrobotics/rdk-devenv:amd64
+              options: --platform linux/amd64
+            artifact-name: module-linux-amd64
+          - platform: darwin/arm64
+            os: macos-latest
+            artifact-name: module-darwin-arm64
+
+    runs-on: ${{ matrix.os }}
+    container: ${{ matrix.container }}
+
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      - name: Set up build environment
+        run: |
+          make setup
+
+      - name: Build module
+        run: |
+          make module.tar.gz
+
+      - name: Upload module artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.artifact-name }}
+          path: module.tar.gz
+
+  publish:
+    needs: build
+    runs-on: ubuntu-latest
+    container:
+      image: ghcr.io/viamrobotics/rdk-devenv:amd64
+      options: --platform linux/amd64
+    strategy:
+      matrix:
+        include:
+          - platform: linux/amd64
+            artifact-name: module-linux-amd64
```

Comment:
why do we name these differently on different platforms?

Reply:
so the publish step can know which artifact from the build step to download and upload